### ISSUE TITLE
fxtwitter Issues

### ISSUE THREAD

**Issue by saxamaphone69** [TuxedoTako/4chan-xt#60]
Personally I've never had any issues with Twitframe, so not sure why this change was forced. I think most of these issues stem from the floating embed option

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/ec2f8ce1-f9fc-4e11-9d68-93d65273a52a)

The setting appears incomplete?

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/182b9f19-63d7-4a17-959d-4344603b7675)

Not even sure what 'resolve' means?

Depending on the content of the tweet, clicking `(embed)` on a tweet then fills up my entire browser and I need to click `Esc` just to close (before I didn't have to)

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/045e788c-fbf5-4914-9aa6-4b89222d3127)

Or they are too long

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/3d959b47-2187-4021-8f06-8064829e258f)

Originally on twitframe I could at least click on the page, any links, and interact

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/9d99643f-abbf-4a45-8c9f-0cdb293f0c7a)

I understand it's meant to be simpler, and I guess inline it looks fine enough

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/83ac1237-cdd5-4ff1-b1ab-84c706b37c1a)

Inspecting the element I see a `[data-options="undefined"]`, so something has gone wrong there (no errors in console)

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/fbbbb5a7-2d9d-4bfd-aaaa-2ead248e3909)

And other than the height issue (which I've addressed in my userstyle), I kinda like how the tweet itself was it's "own" element/card.

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/008085b6-7249-449d-924f-efa5d93eb471)

Just my two cents

**Comment by TuxedoTako** [comment]
> Personally I've never had any issues with Twitframe, so not sure why this change was forced.

It's not forced. If you turn off "Embed Tweets inline with fxTwitter", it uses twitframe. 

The reason for this change is that I got a pull request from someone claiming to have problems with twitframe: https://github.com/TuxedoTako/4chan-xt/pull/57, and I don't get PR's often, so I didn't just want to dismiss or nitpick the few that I get.

@irystocratanon, what are your thoughts on this?

**Comment by irystocratanon** [comment]
I can't look into this right now (probably not until Saturday at the earliest) but I guess it wouldn't hurt to keep the default on false for now.

**Comment by irystocratanon** [comment]
@saxamaphone69 can you share the CSS you added?

I think the way forward would be to set a fixed height and make the blockquote scrollable.

It should also be possible to make it look more like a card and less like something embedded inline.

Links should be clickable already, but we could add extra code to link @ user accounts and hashtags too.

**Comment by saxamaphone69** [comment]
>not forced

More in the sense that I was using twitframe, and my default was switched to fxtwitter. Yes I can switch it back, I know, and have. In the "this change on the default 4chan XT experience did not meet parity of the previous feature it was trying to replace"-sense.

>pr merge

I agree, are great. Unfortunately with the way 4chan X/XT is, there are a lot of options to consider. Working with a variety of different settings, working across boards types, working across domains, accounting for people blocking Twitter or other requests who may need to add new exceptions, OneeChan support.

>off by default

If it is a better replacement than twitframe, I'd be happy to switch. Again it's more because my workflow was changed.

>css

It's somewhere [around here](https://github.com/saxamaphone69/ss21/blob/main/ss21.user.styl#L5192), but I think this is only for the floating embed. I also remove all 4chan XT CSS so there's that too. Doesn't actually _fix_ fix the issue, I think the height being grabbed and the interaction with the `iframe` (in the original code) is the issue.

It would probably be a case of all the content of the tweet being chucked in a `div` and then `max-height` and `max-width`, with an `overflow-y: scroll` or something.
Card-style would be a personal preference, not something I think should be default. I can see the appeal of it looking like the tweet was part of the post, and also see the appeal of it being self-contained. Would need to see an example of when someone links a tweet mid-sentence, or several tweets one after the other.
Middle clicking the profile picture/username or hashtag or even the X logo to be open more info in a new tab is handy. Plus you can click the heart to be taken to the tweet to like it when signed in, reply starts a reply in a new tab, and clicking 'view replies' takes you to the original tweet too.

Overall, I'm not against the change, just need a bit more convincing.

**Comment by saxamaphone69** [comment]
~~Interestingly enough, I just got a message that all the URLs are going to be updated to x.com now, so wonder if that might affect this too.~~
Already handles it!

Other things now that I'm playing around

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/4581f10f-0f45-40a2-a770-324cfc83dd44)
When a link is the last part of the tweet, the dash and `hr` are actually part of the link, meaning the link doesn't work.
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/49bb1edb-f467-4cbf-8669-5ef96d697330)

Quote tweets include profile picture and other clickable bits
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/340cb994-ec40-455d-b6b2-2a571bc2a594)

Got something in the console for that `undefined`
```
SyntaxError: Unexpected token 'N', "Not Found" is not valid JSON
    at JSON.parse (<anonymous>)
    at Object.onload (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:18766:35)
    at wt (<anonymous>:9:89)
    at s.onload (<anonymous>:27:260)
    at wt (<anonymous>:9:89)
    at <anonymous>:36:335
    at wt (<anonymous>:9:89)
    at r (<anonymous>:24:484)
    at <anonymous>:25:112
    at <anonymous>:76:491
```

Probably intentional, but tweets within a `.quote`, as expected, also turn olive.
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/73e16253-3eb8-4130-a7c9-bdc7ad65e70e)

Taking a hunch but the translate feature might also be flawed in some way (`und` as a language, for a full stop)
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/2616118e-0e87-4271-90ce-a5db2e47bf2c)

Videos in tweets preload, which I'm fine with, but maybe some way of disabling it? Not sure if that would be from 4chan XT or fxtwitter's side though. Not sure if this URL helps finding the `poster`.
`https://pbs.twimg.com/ext_tw_video_thumb/1790653231661760512/pu/img/dMbpMkS6fFi-pus2.jpg` or even 
`https://pbs.twimg.com/ext_tw_video_thumb/1790653231661760512/pu/img/dMbpMkS6fFi-pus2?format=webp&name=tiny`

Unsure if 4chan XT or fxtwitter is dictating the HTML structure, but things like the username/time being wrapped in its own element and the 'stats' being wrapped in an element would make things easier to style and sort with flexbox.

![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/0e98d6aa-bddf-4a85-a9b8-4ab47d8218e5)
Tweets with multiple images appear all in a row, so unsure what would happen with 4. I also assume the `no-referrer` should be on the `a` not the `img`?

I might be going crazy, but does it not support polls with images?
https://x.com/MKBHD/status/1203439521179820032

**Comment by TuxedoTako** [comment]
> Unsure if 4chan XT or fxtwitter is dictating the HTML structure, but things like the username/time being wrapped in its own element and the 'stats' being wrapped in an element would make things easier to style and sort with flexbox.

4chan XT builds the html in src/Linkification/Embedding.tsx. The pull request used template strings, but I changed it to jsx to make sure everything is escaped.

**Comment by saxamaphone69** [comment]
This probably helps in absolutely no way, but I (ChatGPT + I) mucked around with [some ideas on Codepen](https://codepen.io/saxamaphone69/pen/BaejLLj) as to what the card idea could be. Though admittedly, I like how embed tweets look already, so essentially just recreated that look.

**Comment by aidanharris** [comment]
> This probably helps in absolutely no way, but I (ChatGPT + I) mucked around with [some ideas on Codepen](https://codepen.io/saxamaphone69/pen/BaejLLj) as to what the card idea could be. Though admittedly, I like how embed tweets look already, so essentially just recreated that look.

This looks very nice. I happen to like the inline look but I can see the appeal of something like this. I wonder if we can make a re-usable web component out of this (ideally without pulling in all of React, some quick searching makes me think this should be possible with just jsx).

This could better contain the logic and would also make it easier to style it if you use custom styles and would also serve as a nice example of how we can split out embedding into separate modules instead of one big Embedding.tsx.

Just a thought.

**Comment by irystocratanon** [comment]
> This probably helps in absolutely no way, but I (ChatGPT + I) mucked around with [some ideas on Codepen](https://codepen.io/saxamaphone69/pen/BaejLLj) as to what the card idea could be. Though admittedly, I like how embed tweets look already, so essentially just recreated that look.

This looks very nice. I happen to like the inline look but I can see the appeal of something like this. I wonder if we can make a re-usable web component out of this (ideally without pulling in all of React, some quick searching makes me think this should be possible with just jsx).

This could better contain the logic and would also make it easier to style it if you use custom styles and would also serve as a nice example of how we can split out embedding into separate modules instead of one big Embedding.tsx.

Just a thought.

**Comment by saxamaphone69** [comment]
Without looking too much into it, you mean essentially recreating this? https://www.webcomponents.org/element/twitter-status
I'm not too familiar with React/JSX. My mind struggles enough with template literals (which is what I used in that example... I think?) 

If you comment out all the CSS, other than the huge profile pictures and media, it looks pretty much inline, like it is now. As long as things are wrapped in elements with discernible class names, people can style them however they want. Add a section to the wiki for different style options (default inline, twitter card, make it look like its own 4chan post) that people can add to their own Custom CSS, as to not add bloat within 4chan XT itself.

```
twitter card
  ?(twitter reply)
  twitter text
  ?(twitter translate)
  ?(twitter rich content)
    ?(twitter poll | twitter media)
  twitter info
  twitter stats
```

Maybe all the extra settings, after the initial "switch to fxTwitter", can be put into `Advanced` instead? There you could add the translation options, quote tweet options, options for media or polls, or even themes. These can just be extra classes added to the main `twitter-card` parent.

Main focus I think should be fixing bugs and ensuring all options are supported out of the box. The foundation/base needs to be solid first.

**Comment by TuxedoTako** [comment]
I started a new branch based on the codepen: fxtwitter-improvements. It's work in progress at the moment (see the commit message for to do's, it has old code in comments or after a return), but suggestions are already welcome.

**Comment by irystocratanon** [comment]
> I'm not too familiar with React/JSX. My mind struggles enough with template literals (which is what I used in that example... I think?)

Template literals are confusing, yes. I did an experiment but it doesn't look good:
https://github.com/TuxedoTako/4chan-xt/compare/project-XT...irystocratanon:4chan-xt:fxtwitter-webcomponent

The <style> didn't work properly at all in several places without adding extra inline styles. It looks more like a card though.

**Comment by saxamaphone69** [comment]
I don't mind the web component idea personally, I guess it depends on what precedent it might start for other parts of 4chan XT.

Keep in mind that my CSS was merely a suggestion, it was in no means perfect or even covers all potentials (e.g. 3 media pieces). I used some relatively new syntax like nested CSS, `:is()`, `:has()`, and `color-mix` which might be causing things to not work? Does `:host` come into play? Custom properties can just be set on `.twitter-card` as opposed to `:root`.

Ideally `@scope` comes to mind too as something that would be handy here, if it had Firefox support. I also thought that `scroll-snap` with an `overflow-x` could be nice for tweets with 4 media pieces.

Sadly, I am not smart enough to see what either branch actually looks like in practice (how to build etc) or how to contribute any more than my isolated Codepen example.

**Comment by irystocratanon** [comment]
> I don't mind the web component idea personally, I guess it depends on what precedent it might start for other parts of 4chan XT.

My thinking was it'd be easier to style and modify just by changing the template but either I am misusing templates (likely) or I don't know enough about how to make this work properly.

If it worked well it'd be more composable but I don't know enough here.

I agree with @TuxedoTako it's better to just fix the issues.

**Comment by irystocratanon** [comment]
I added back the translation in https://github.com/irystocratanon/4chan-xt/commit/763f1bfb69ab6a3b373dd1310e954b12f740012b this should work like it was supposed to:

If both a Tweet and Quote have translations then it groups them both at the top (excluding media). Otherwise the translation appears below the Tweet/Quote.

**Comment by TuxedoTako** [comment]
> > I'm not too familiar with React/JSX. My mind struggles enough with template literals (which is what I used in that example... I think?)
> 
> Template literals are confusing, yes. I did an experiment but it doesn't look good: [project-XT...irystocratanon:4chan-xt:fxtwitter-webcomponent](https://github.com/TuxedoTako/4chan-xt/compare/project-XT...irystocratanon:4chan-xt:fxtwitter-webcomponent)
> 
> The <style> didn't work properly at all in several places without adding extra inline styles. It looks more like a card though.

What do we gain from wrapping it into a custom component over just putting it into the DOM? The css classes are prefixed, so those shouldn't conflict anyway. I actually prefer the embedded tweet to match the chosen 4chan theme colors. 

> Keep in mind that my CSS was merely a suggestion, it was in no means perfect or even covers all potentials (e.g. 3 media pieces). I used some relatively new syntax like nested CSS, :is(), :has(), and color-mix which might be causing things to not work? 

Now that you mention it, I already removed color-mix and nesting, but :has() isn't supported by our current minimum versions. Those can be bumped, but I don't know by how much. I remember a github issue from someone using Firefox ESR, which is a bit behind.

> Does :host come into play?  Custom properties can just be set on .twitter-card as opposed to :root.

By the way, I changed the .twitter- prefix to .fxt- in my code. I also prefixed the custom properties with --xt-fxt-, so those shouldn't conflict anyway.

> Sadly, I am not smart enough to see what either branch actually looks like in practice (how to build etc) or how to contribute any more than my isolated Codepen example.

Building should be as simple as installing node, running an `npm install`, and then running a build script from package.json. You can even run them with just one click from vs code: 

![Code_NszXOQz8Wj](https://github.com/TuxedoTako/4chan-xt/assets/126348645/19216c41-a529-44f9-9f35-232793dca767)

If you're using a different editor, you can just run `npm run build` in the terminal. The build just overwrites those in the /builds directory. But please don't put those builds in a PR, I only commit a new build when I create a new release.

[There are some more build options in the readme](https://github.com/TuxedoTako/4chan-xt?tab=readme-ov-file#build-from-source), but you don't need to worry about those, I use the defaults myself.

> I added back the translation in https://github.com/irystocratanon/4chan-xt/commit/763f1bfb69ab6a3b373dd1310e954b12f740012b

I'll look at that later.

**Comment by irystocratanon** [comment]
> What do we gain from wrapping it into a custom component over just putting it into the DOM? The css classes are prefixed, so those shouldn't conflict anyway. I actually prefer the embedded tweet to match the chosen 4chan theme colors.

Not a lot as far as I can tell.

I rebased off of your fxtwitter-improvements branch and think I fixed most of the issues now if you want to pull from https://github.com/irystocratanon/4chan-xt/tree/fxtwitter-improvements

I didn't make a pull request because you still might want to make other changes.

**Comment by TuxedoTako** [comment]
Released some improvements in [2.10.0](https://github.com/TuxedoTako/4chan-xt/releases). There's probably still room for improvement, but at least it doesn't look completely unfinished.

**Comment by saxamaphone69** [comment]
Looking a lot better! List of things I noticed that still need addressing (just a mental note, rather than demands, I _might_ try a pass at some of these)

- [ ] still getting the `undefined` error (only with fxtwitter option)
```
userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:18836 SyntaxError: "undefined" is not valid JSON
    at JSON.parse (<anonymous>)
    at Object.onload (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:18834:35)
    at wt (<anonymous>:9:89)
    at s.onload (<anonymous>:27:260)
    at wt (<anonymous>:9:89)
    at <anonymous>:36:335
    at wt (<anonymous>:9:89)
    at r (<anonymous>:24:484)
    at <anonymous>:25:112
    at <anonymous>:76:491
```
- [ ] not sure if the `width: fit-content` stuff is actually doing anything
- [ ] floating embed still an issue (something needs a `max-height` and an `overflow-y: auto`, seems like a lot of empty div nesting for no benefit)
- [ ] i feel like the padding on the card might be too much
- [ ] i feel like the `48px` for the avatar might be too big
- [ ] name and @ might be better with an `align-items: center`
- [ ] rogue `console.log` 
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/432f0611-1df2-442a-8d17-3c6b21c07700)
- [ ] `fxt-text` is in a `div`, but translated text goes back to `p`, should be consistent probably
- [ ] speaking of translated text, would be good if the text itself is wrapped and not a straggling child of `fxt-text`, for greater style options
- [ ] speaking of straggling child, the `stats_meta` not being aligned is kinda bugging me
- [ ] `fxt-media_container` is still rendered even when there is no media
- [ ] `fxt-media_container ` has a trailing space
- [ ] tomorrow and photon `hr` is the same colour as the post reply, change it to `.fxt-card hr { border-color: var(--xt-fxt-border) }`
- [ ] long usernames could be a problem 
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/51f19dea-59a3-4290-849b-e7d1ead61553)
- [ ] that `aspect-ratio` i used on avatars... i dont think it even works lol
- [ ] quote tweets need some breathing room/space (see next pic)
- [ ] gallery with 3 pics doesnt quite look right... maybe the mosaic can just be used instead but also individual pics feels better 
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/aa18ad33-7bb9-45bf-9a8f-89502587191e)
- [ ] on initial load, still appears like you can resize when you cant
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/4440aefa-f907-4042-9df2-3d215cc1b308)
- [ ] "replying to" option doesnt load previous tweets (i dont think the code is there yet) (probably should max it out at 5 not 100) (as for how it looks... maybe some `details` and `summary` elements might help to collapse elements otherwise vertically that would take a lot of space)

**Comment by TuxedoTako** [comment]
> still getting the undefined error (only with fxtwitter option)

I have error handling for 404, but not other errors. The undefined error hasn't happened on my machine.

> not sure if the ´width: fit-content´ stuff is actually doing anything

it resizes the div to it's contents. The embed is in a div because that's how they were implemented, and I don't want to change a working implementation.

> floating embed still an issue (something needs a max-height and an overflow-y: auto, seems like a lot of empty div nesting for no benefit)

Forgot to test that.

> "replying to" option doesnt load previous tweets (i dont think the code is there yet) (probably should max it out at 5 not 100) (as for how it looks... maybe some details and summary elements might help to collapse elements otherwise vertically that would take a lot of space)

Should be fixed.

---

The rest is kind of subjective. I already underestimated the time I would spend on the new embed, so I don't feel like putting more work in that. Pull requests are still welcome.

**Comment by Hash6232** [comment]
Since it's related to fxtwitter I'll post this here. Embedding a tweet containing a video in portrait mode will force the dialog to go out of bound depending on the size of your screen. As a test, I tried adding a `max-height` property to `.fxt-media > video` and it appears to be working as intended. Maybe you should consider adding max sizes limits for tweets with media so that the dialog doesn't become completely unusable due to the topbar moving out of bound

![ezgif com-animated-gif-maker](https://github.com/TuxedoTako/4chan-xt/assets/164906916/56c58d00-a44f-431f-be6f-baafd3fad95b)

**Comment by AlifianK** [comment]
![image](https://github.com/TuxedoTako/4chan-xt/assets/50392169/a1b6b707-ea40-4dab-b1c2-25769f2df7a8)
Is the switched reply and like counter a fxtwitter issue or 4chanXT?

**Comment by TuxedoTako** [comment]
> Is the switched reply and like counter a fxtwitter issue or 4chanXT?

4chanXT, and it's a too big of an issue to procrastinate. So, here's [2.10.3](https://github.com/TuxedoTako/4chan-xt/releases).

---

Unrelated to that issue: something else nice to have related to the fxTwitter embeds is community note support.

**Comment by Hash6232** [comment]
Wrapping a twitter link in a spoiler will also spoiler the tweet text and UI

![image](https://github.com/TuxedoTako/4chan-xt/assets/164906916/27eaaf24-ff9d-4cb3-9cc2-5581bc7c3986)

**Comment by saxamaphone69** [comment]
On that point, might need to handle dead tweets too:
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/0d5886eb-cb21-43a3-a526-f33d0362446f)

**Comment by Katzenwerfer** [comment]
> The reason for this change is that I got a pull request from someone claiming to have problems with twitframe: #57, and I don't get PR's often, so I didn't just want to dismiss or nitpick the few that I get.

From what I could find, some twitframe issues are caused by Firefox's tracking protection. Unless problems with it also occur on Chromium-based browsers, which in that case I wouldn't know.

**Comment by Hash6232** [comment]
> From what I could find, some twitframe issues are caused by Firefox's tracking protection. 

Adding `boards.4chan.org` to the enchanced tracking protection exceptions did fix images not showing at all on my end. So that setting can definitely affect the embedder considering that by default the strict mode is akin to a blanket rule that hinders all socials on third party websites

**Comment by saxamaphone69** [comment]
Not that it helps much (given that I did it on CodePen rather than on 4chan/with 4chan XT, but I wasn't going to wait 900 seconds to do a bunch of Twitter tests) but I spent some time this weekend looking over the `fxtwitter` stuff.

https://codepen.io/saxamaphone69/full/dPbjqNP

Once again, ChatGPT helped me out with the JavaScript.

I did [post a tweet](https://boards.4chan.org/vg/thread/510570176#p510674263) that I know has replies, and I have it enabled (see below) but replies don't show for me.
![Image](https://github.com/user-attachments/assets/2c9202e2-2f36-465c-98c2-dd059019fb38)

Probably a few edge cases to address still. For example, I'm going to assume if a quote tweet or a reply tweet that is deleted may not work (like: `https://x.com/ViLettuce/status/1654324865774637057`). Also not sold on the styling - would need to start it again and ensure it cascades correctly (and keep all the numbers and colours consistent in the `.fxt` scope.

**Comment by TuxedoTako** [comment]
> Not that it helps much (given that I did it on CodePen rather than on 4chan/with 4chan XT, but I wasn't going to wait 900 seconds to do a bunch of Twitter tests) 

You're overcomplicating this. You have access to the source code. You can load any tweet you want without having to post them.

Try this for example:

```patch
diff --git a/src/Linkification/Embedding/FxTwitter.tsx b/src/Linkification/Embedding/FxTwitter.tsx
index 927bbe46f..6e50e04e5 100644
--- a/src/Linkification/Embedding/FxTwitter.tsx
+++ b/src/Linkification/Embedding/FxTwitter.tsx
@@ -9,7 +9,8 @@ import { Conf } from '../../globals/globals';
 export default function EmbedFxTwitter(a: HTMLAnchorElement): HTMLElement {
   const el = $.el('div', { innerHTML: '<blockquote class="twitter-tweet">Loading&hellip;</blockquote>' });
   const shouldTranslate = Conf.fxtLang ? `/${Conf.fxtLang}` : '';
-  CrossOrigin.cachePromise(`${Conf.fxtUrl}/${a.dataset.uid}${shouldTranslate}`).then(async req => {
+
+  CrossOrigin.cachePromise(prompt("FxTwitter url", `${Conf.fxtUrl}/${a.dataset.uid}${shouldTranslate}`)).then(async req => {
     if (req.status === 404) {
       el.textContent = '404: tweet not found';
       return;
```

This one line change will prompt you for an URL any time you open an embed, and uses that URL instead.

I already tried to make changes based on your codepen. Those aren't released yet, but they are already in the project-XT branch if you want to try them. Please just make a git branch for future changes. It will be the easiest for both of us.

**Comment by saxamaphone69** [comment]
Thank you @TuxedoTako - it wasn't my intention to create more work for you, sorry.

My holiday break is coming to a close, so any contributions I do make will be small (as to not break things again) but I will endeavour to try and make them using the proper methods/channels.

**Comment by saxamaphone69** [comment]
~~Okay yeah, just built it and there are some things that need some work now that it's part of 4chan - including some icon alignment tweaks from the SVG PR (I'll open a new issue).~~
I think I'd also like to revisit other points raised in this issue to make sure they've all been accounted for.

Other things:
- hashtags don't seem to work for non-english tweets
- change poll to ```<div class="fxt-total-votes">
				<span class="fxt-total-votes_number">${poll.total_votes.toLocaleString()} votes</span>
				<span class="fxt-total-votes_spacer">·</span>
				<span class="fxt-total-votes_time">${poll.time_left_en.toLocaleString()}</span>
			</div>``` to include how much longer an active poll has, and "Final results" when it is ended
- for translations, perhaps show the translated text first (e.g. English) and then, the translation is hidden in a `<details>` and `<summary>` which can be expanded "View original text in `${language}`"
- `body`, `a`, `a:hover` colour might also need to be stored in a css custom property to avoid the `.quote` and `s / .spoiler` inheritance issue (so it can be redeclared on `.fxt-card`) OR the other option is when embedding finds itself inside an element, it appends itself to the end of the parent element.

So, instead of this, `media-embed` inserts itself, in this situation, between the closing `span` and `br`.  I can't imagine if someone wrote `>hello x.com hello` that the fact the embed would be after the second hello would actually matter. If anything, it would look better?
```
<blockquote class="postMessage" id="m94523433">
<span class="quote">
&gt;<a class="linkify twitter" rel="noreferrer noopener" target="_blank" href="https://x.com/vanilla7s26/status/1883754586638540912">https://x.com/vanilla7s26/status/1<wbr>883754586638540912</a> <a class="embedder embedded" href="javascript:;" data-key="Twitter" data-uid="vanilla7s26/status/1883754586638540912" data-options="undefined" data-href="https://x.com/vanilla7s26/status/1883754586638540912">(<span>un</span>embed)</a>
<div class="media-embed">
<div style="border: none;" class="fxt-card_container">
</div>
</div>
</span>
<!-- here instead -->
<br>
someone is bullying her
</blockquote>
```

**Comment by TuxedoTako** [comment]
I broke `--xt-fxt-fg` in 38a20d1c6787f417620a893996228d802a6e0088.
